### ISSUE TITLE
WinUI 3 Preview 4: The target process exited without raising a CoreCLR started event

### ISSUE THREAD

**Issue by thomasclaudiushuber** [microsoft/microsoft-ui-xaml#4226]
**Describe the bug**

The blank WinUI 3 Preview 4 desktop application does not start.
The output window shows this error: 

_The target process exited without raising a CoreCLR started event. Ensure that the target process is configured to use .NET Core. This may be expected if the target process did not run on .NET Core.
The program '[6384] App15.exe' has exited with code -2147450740 (0x8000808c)._

**Steps to reproduce the bug**
1. Create a new "Blank App (WinUI in Desktop)"
2. Add a reference to the NuGet Package Entity
```
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.3" />
```
3. Try to debug the app. It builds, gets deployed, but gets not started
4. The output window shows the mentioned error

**Expected behavior**
The application should start after adding a reference to the NuGet package Microsoft.EntityFrameworkCore.SqlServer

**Version Info**
WinUI 3 Preview 4 (but also happens on Preview 3).
I include here my .csproj for details
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net5.0-windows10.0.18362.0</TargetFramework>
    <TargetPlatformMinVersion>10.0.17134.0</TargetPlatformMinVersion>
    <RootNamespace>App15</RootNamespace>
    <ApplicationManifest>app.manifest</ApplicationManifest>
    <Platforms>x86;x64;arm64</Platforms>
    <RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.3" />
    <PackageReference Include="Microsoft.WinUI" Version="3.0.0-preview4.210210.4" />
    <Manifest Include="$(ApplicationManifest)" />
  </ItemGroup>
</Project>

```

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  | Yes |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        |    |
| May 2020 Update (19041)            |  Yes   |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Comment by thomasclaudiushuber** [comment]
Seems to be NuGet related. (cc @rrelyea)

When I add EntityFrameworkCore in addition (which is referenced by EntityFrameworkCore.SqlServer) to the WinUI project, then it works. This .csproj entry in WinUI makes the error go away:
```
  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.3" />
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.3" />
    <PackageReference Include="Microsoft.WinUI" Version="3.0.0-preview4.210210.4" />
    <Manifest Include="$(ApplicationManifest)" />
  </ItemGroup>  
  ```

**Comment by marb2000** [comment]
Hey @thomasclaudiushuber I couldn't repro it. It works well on my two machines. Did you have installed other .NET Frameworks like preview of .NET 6 or something like that?

**Comment by thomasclaudiushuber** [comment]
Hi @marb2000 ,

thanks for your answer and for trying to reproduce it.

I also tried it on another machine, and I also could reproduce it there (Seems our machines are mutually exclusive ☺) I originally got this error reported from a developer who watched a video from me, then they send me their code, and I couldn't get it working. I investigated into it, and found out that EF SqlServer NuGet package produces the error. And I think EF Core SqlServer could be an important package.

There's nothing special on my two machines, I have installed VS Community and VS Community Preview (and the .NET stuff coming with these)

```
C:\Users\thoma>dotnet --list-sdks
2.2.204 [C:\Program Files\dotnet\sdk]
3.0.100 [C:\Program Files\dotnet\sdk]
3.1.406 [C:\Program Files\dotnet\sdk]
5.0.103 [C:\Program Files\dotnet\sdk]
5.0.200-preview.21079.7 [C:\Program Files\dotnet\sdk]
```

```
C:\Users\thoma>dotnet --list-runtimes
Microsoft.AspNetCore.All 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.All 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.All 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.0.0-preview8.19405.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.0.0-preview9.19424.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.1.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 2.2.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.0.0-preview9-19423-09 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.0.0-preview9-19423-09 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

The issue happens in both machines, both have similar configurations.

Installed Visual Studio versions:
VS Community 16.9.0 Preview 4
VS Community 16.8.5

Error is reproducable in both instances for me.

But it only occurs with this package:
```
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.3" />
```

Referencing for example inMemory instead
```
    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="5.0.3" />
```
makes the application working after a rebuild.

I can't get the application working with the Microsoft.EntityFrameworkCore.SqlServer NuGet package. But for example a .NET 5 WPF app with that NuGet package works fine.

**Comment by StephenLPeters** [comment]
@marb2000 and @alwu-msft FYI

**Comment by alwu-msft** [comment]
This is the (tail end of the) output I see in the debugger when attempting to launch a blank app that references the Microsoft.EntityFrameworkCore.SqlServer Nuget package:
```
ModLoad: 76030000 76055000   C:\WINDOWS\SysWOW64\IMM32.DLL
ModLoad: 71870000 718d0000   C:\Program Files (x86)\dotnet\host\fxr\5.0.3\hostfxr.dll
ModLoad: 70bd0000 70c30000   C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App\5.0.3\hostpolicy.dll
Error:
  An assembly specified in the application dependencies manifest (App1.deps.json) was not found:
    package: 'Microsoft.Data.SqlClient.SNI.runtime', version: '2.0.1'
    path: 'runtimes/win-x86/native/Microsoft.Data.SqlClient.SNI.pdb'ModLoad: 76870000 768cd000   C:\WINDOWS\SysWOW64\bcryptPrimitives.dll
ModLoad: 750e0000 750ef000   C:\WINDOWS\SysWOW64\kernel.appcore.dll
eax=00000000 ebx=00000000 ecx=00000004 edx=00000000 esi=8000808c edi=77875b40
eip=777c19bc esp=005ffc54 ebp=005ffd2c iopl=0         nv up ei pl nz na po nc
cs=0023  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00000202
ntdll!NtTerminateProcess+0xc:
777c19bc c20800          ret     8
```
This doesn't look like a WinUI problem per se; rather, I think it's a problem with the WAP project system. `Microsoft.Data.SqlClient.SNI.dll` appears to be getting copied to the package root by the `.wapproj` build rather than `<root>/runtimes/win-x86/native/` as the `.deps.json` expects.

@thomasclaudiushuber Could you report this issue via Visual Studio's [**Help > Send Feedback > Report a Problem.**](https://docs.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2019) and include the information I wrote above? That should get your issue routed to the right area; feel free to ping me if it languishes and I can try to get it some traction.

**Comment by alwu-msft** [comment]
Update: this is almost certainly an issue with the WAP project system (or maybe how the `Microsoft.Data.SqlClient.SNI.runtime` Nuget package is authored). I created a .NET 5 WPF blank app, added a reference to the Nuget package, added a Windows Application Packaging project that wrapped the WPF app, and encountered the exact same error.

**Comment by thomasclaudiushuber** [comment]
Thank you @alwu-msft. Omg, you're right, why the heck didn't I try the WAP project with a .NET 5 WPF project?! :-)

Yes, I can confirm, it's an issue that is related to the WAP project and the NuGet package. I was able to reproduce it in a .NET 5 WPF app when packaging it as MSIX via WAP project.

I reported the error via Visual Studio Send Feedback as you suggested. I keep this issue open and will update/close it, as soon as I get feedback from the other channel. But if you think closing this issue now is a better idea, feel free to close it.

PS: How did you get the detailed debug output? Via WinDbg?
